src/main/java/com/google/step/YOUR_PROJECT_NAME_HERE/servlets/SearchServlet.java

```diff
@@ -15,24 +15,100 @@
 package com.google.step.YOUR_PROJECT_NAME_HERE.servlets;
 
 import java.io.IOException;
-import java.util.Arrays;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.ServletException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import org.json.JSONObject;
```

Comment:
Thinking forward a bit - at some point you'll probably be sending down JSON for the knowledge cards. In the portfolio projects you used GSON for this. I think org.json works too, but maybe think about whether you'd want to use it for producing JSON as well - it'd be better to use only one JSON library.